fmom/pxpypze.go

```diff
@@ -211,3 +211,17 @@ func (p4 *PxPyPzE) Set(p P4) {
 	p4[2] = p.Pz()
 	p4[3] = p.E()
 }
+
+func dotProduct(p1, p2 *PxPyPzE)float64{
+	dotproduct := p1.Px()*p2.Px()+p1.Py()*p2.Py()+p1.Pz()*p2.Pz()
+	return dotproduct
+}
+
+//return the cosine of the angle between the momentum of two jets
+func CosTheta(p1, p2 *PxPyPzE) float64 {
```

Comment:
rather than taking two `*PxPyPzE`, perhaps take 2 `P4` interfaces instead.
it may be slower but more general :) (and push comes to shove, we may implement fast-path later on)